src/main/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTask.java

```diff
@@ -85,24 +86,44 @@ void stop() {
   /** @see MongoSinkTask#put(Collection) */
   void put(final Collection<SinkRecord> records) {
     if (lastTaskInvocation != null) {
-      statistics.timeSpentOutsidePutTask(lastTaskInvocation);
+      statistics
+          .getBetweenTaskInvocations()
+          .sample(lastTaskInvocation.getElapsedTime(TimeUnit.MILLISECONDS));
     }
-    Timer taskTime = statistics.putTaskInvoked();
-    statistics.recordsReceived(records.size());
+    Timer taskTime = Timer.start();
+    statistics.getRecordsReceived().sample(records.size());
+    trackLatestRecordTimestampOffset(records);
     if (records.isEmpty()) {
       LOGGER.debug("No sink records to process for current poll operation");
     } else {
-      Timer processingTime = statistics.startTimer();
+      Timer processingTime = Timer.start();
       List<List<MongoProcessedSinkRecordData>> batches =
           MongoSinkRecordProcessor.orderedGroupByTopicAndNamespace(
               records, sinkConfig, errorReporter);
-      statistics.putTaskRecordProcessingTime(processingTime);
+      statistics
+          .getRecordsProcessing()
+          .sample(processingTime.getElapsedTime(TimeUnit.MILLISECONDS));
       for (List<MongoProcessedSinkRecordData> batch : batches) {
         bulkWriteBatch(batch);
       }
     }
-    statistics.putTaskTime(taskTime);
-    lastTaskInvocation = statistics.startTimer();
+    statistics.getTaskInvocations().sample(taskTime.getElapsedTime(TimeUnit.MILLISECONDS));
+    lastTaskInvocation = Timer.start();
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(statistics.getName() + ": " + statistics.toJSON());
+    }
+  }
+
+  private void trackLatestRecordTimestampOffset(final Collection<SinkRecord> records) {
+    OptionalLong latestRecord =
+        records.stream()
+            .filter(v -> v.timestamp() != null)
+            .mapToLong(ConnectRecord::timestamp)
+            .max();
+    if (latestRecord.isPresent()) {
+      long offsetMs = System.currentTimeMillis() - latestRecord.getAsLong();
```

Comment:
I guess there is nothing we can do about the non-monotonicity of currentTimeMillis, if we want to track this stat?  Should we at least check for negative values here and ignore them?

Reply:
Negative values seem fine, because if one clock is behind the other, we might start with a stable `-10000`, and perhaps set an alert if it moves up by 5000ms, signalling a lag spike. If we capped the negatives, any change would be obscured.

I imagine that in cases where non-monotonicity comes into play, such as a fast/slow clock being reset, we would want to see the reset reflected here.

Reply:
I actually have no idea where the timestamp in ConnectRecord comes from.  Could it be from a different process or server?

Reply:
The timestamp type associated with the timestamp is either the "create time" or the "log append" time. I don't know what it could be if not a reasonable approximation of the time the record was put into Kafka (and there's no other field we might use instead). [This SO answer and comment](https://stackoverflow.com/a/62936145) suggest that the default is "local epoch time of the producing system".

Reply:
Yeah, then not a lot we can do here.  I'll resolve